src/pages/Search/__test__/Search.test.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { Route, MemoryRouter } from 'react-router-dom';
+import {
+  render,
+  cleanup,
+  waitForElementToBeRemoved,
+  fireEvent,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { ThemeProvider } from 'styled-components';
+import axiosMock from 'axios';
+import Search from '../index';
+import Theme from '../../../styles/theme';
+
+jest.mock('axios');
+afterEach(cleanup);
+
+const renderWithRouter = (component, subreddit) => ({
+  ...render(
+    <ThemeProvider theme={Theme}>
+      <MemoryRouter initialEntries={[`/search/${subreddit}`]}>
+        <Route path="/search/:subreddit">{component}</Route>
+      </MemoryRouter>
+    </ThemeProvider>,
+  ),
+});
+
+const fakeData = () => {
+  const post = {
+    title: 'testing',
+    full_link: 'link',
+    created_utc: 1586823316,
+    score: 50,
+    num_comments: 50,
+    author: 'Ayoub',
+  };
+  const hours = Array.from({ length: 24 }, (v, i) => [i === 0 ? post : { i }]);
+  const data = Array.from({ length: 7 }, () => hours);
+  return data;
```

Comment:
If this is the mock response data it should be an array of posts. This looks like the converted data for the heatmap already

Reply:
You mean I should just send an array of posts? and let convertedData.js do the job?

Reply:
Yes, exactly. It's totally correct, that you mock the `axios.get` function since this is what sends the API request. But it should return the same response as it usually would do. You can check the network tab of your developer tools to see what that response can look like (see the screenshot). You can also copy a part of that response and use it as mock data. Since mock data can be very long you can save it into a separate `.json` file in the `__tests__` folder and import it in the test file.

![image](https://user-images.githubusercontent.com/9722792/79253093-f3d6a480-7e82-11ea-9b05-3b073f1bd774.png)

Reply:
will do, Thank you very much

Reply:
I only saved data that I use (title,created_utc.link .....) I don't know if this okay or should save the whole post  object